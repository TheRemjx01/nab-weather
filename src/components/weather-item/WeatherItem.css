.weatherItem {
  width: calc(160px - 8px);
  padding: 4px;
  background-color: #202125;
  border-radius: 4px;
}

.weatherItem .icon {
  width: 50px;
  height: 50px;
}

.weatherItem .title {
  font-size: 1.6rem;
}

.weatherItem .tempContainer {
  list-style: none;
  font-size: 1.2rem;
  margin-block-start: 0;
  margin-block-end: 0;
  padding-inline-start: 0;
}
