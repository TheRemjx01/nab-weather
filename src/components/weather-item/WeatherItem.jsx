import React from "react";
import round from "lodash/round";
import PropTypes from "prop-types";

import WeatherIcon from "../weather-icon";
import "./WeatherItem.css";

export const getDisplayTemp = (temp) => `${round(temp, 2)}*C`;

export const WeatherItem = ({ weekDay, minTemp, maxTemp, weatherIcon }) => (
  <div className="weatherItem">
    <h4 className="title">{weekDay}</h4>
    <WeatherIcon weather={weatherIcon} className="icon" />
    <ul className="tempContainer">
      <li>Min: {getDisplayTemp(minTemp)}</li>
      <li>Max: {getDisplayTemp(maxTemp)}</li>
    </ul>
  </div>
);

WeatherItem.propTypes = {
  weekDay: PropTypes.string.isRequired,
  weatherIcon: PropTypes.string.isRequired,
  minTemp: PropTypes.number.isRequired,
  maxTemp: PropTypes.number.isRequired,
};

export default WeatherItem;
