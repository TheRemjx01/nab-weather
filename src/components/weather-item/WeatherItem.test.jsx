import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { WeatherItem, getDisplayTemp } from './WeatherItem'

describe('getDisplayTemp', () => {
 it('should be able to get correct temperature', () => {
  const temp = 29.577;
  expect(getDisplayTemp(temp)).toBe('29.58*C')
 })
})

describe('<WeatherItem />', () => {
 it('should be able to render', () => {
  const Wrapper = shallow(<WeatherItem weekDay='Saturday' minTemp={25.53} maxTemp={28.53} weatherIcon='sn'/>);
  expect(toJson(Wrapper)).toMatchSnapshot();
 })
})

