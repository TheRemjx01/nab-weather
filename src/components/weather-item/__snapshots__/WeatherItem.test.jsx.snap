// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherItem /> should be able to render 1`] = `
<div
  className="weatherItem"
>
  <h4
    className="title"
  >
    Saturday
  </h4>
  <WeatherIcon
    className="icon"
    weather="sn"
  />
  <ul
    className="tempContainer"
  >
    <li>
      Min: 
      25.53*C
    </li>
    <li>
      Max: 
      28.53*C
    </li>
  </ul>
</div>
`;
